integration/grafana/README.md

```diff
@@ -1,25 +1,147 @@
+build and install a development snapshot of GEOPM.
+
+```bash
+    # Install latest development snapshot of GEOPM
+    GEOPM_PREFIX=... # target installation directory
+    wget https://raw.githubusercontent.com/geopm/geopm/refs/heads/dev/geopmdpy/install_user.sh
+    chmod a+x install_user.sh
+    ./install_user.sh --prefix=$GEOPM_PREFIX --enable-levelzero
+```
+
+
+### Configuring ports
+
+There are three ports used by the script to communicate:
+
+- Prometheus server port opened on the head node (default 9090)
+- Grafana server port opened on the head node (default 3000)
+- Prometheus client port opened on the allocated nodes (default 8000)
+
+To avoid conflicts, or firewall limitations, the user may override the defaults
+using command line arguments to pbs_prometheus_launch.py
+
+### Configuring Grafana Server
+
+Before deploying the GEOPM Prometheus exporter across the system for the first
+time, the Grafana server needs to be configured on the head node by setting the
+administrator password and uploading the GEOPM Power Dashboard though the web
+GUI.
+
+To set the administrator password for your Grafana instance with the `grafana-cli`
+tool run the following command inside of the Grafana software directory:
+
+```bash
+    cd $GRAFANA_DIR/
+     ./bin/grafana-cli admin reset-admin-password $GRAFANA_ADMIN_PASSWORD
+```
+
+To add the GEOPM Prometheus dashboard to your Grafana configuration the Grafana
+and Prometheus servers must be running.  This is done by executing the
+`pbs_prometheus_launch.py` script without specifying the `--jobid` option.  This
+will bring up the Prometheus server and Grafana server on the head node.
+
+Then navigate to the Grafana web GUI in a web browser to import the GEOPM
+Dashboard.  The Grafana server will be running an http (not https) server on the
+head node URL on port 3000 (port can be modified with the the --graf-port option
+to `pbs_prometheus_launch.py`).  For example if the head node hostname is
+`login.cluster.acme.com` then the Grafana server is reached through the URL
+`http://login.cluster.acme.com:3000`.  Login into the Grafana web page with the
+`admin` user credentials set previously using the `grafana-cli` tool.  Once
+logged in, add the http Prometheus data source using port 9090 (unless overridden
+with the --prom-port option to `pbs_prometheus_launch.py`),
+e.g.`http://localhost:9090`.  Next, import the [GEOPM Power Report Grafana
+dashboard configuration
+file](https://raw.githubusercontent.com/geopm/geopm/refs/heads/dev/integration/grafana/GEOPM_Report.grafana.dashboard.json)
+using the Prometheus data source by drag-and-drop or copy-paste.
+
+
+### Monitor the user job using the GEOPM Prometheus & Grafana framework
+
+Now that you have configured Grafana and Prometheus, you can launch the Prometheus
+client exporters across the allocated nodes of a job, and then use the Grafana Web GUI
+(running on port `GRAFANA_SERVER_PORT` to monitor the aggregated telemetry.
```

Comment:
Parenthesis not closed. However, this note might not be necessary since the documentation already explained previously how to reach the Grafana GUI. If included, I would recommend to not mention `GRAFANA_SERVER_PORT`, as this is not mentioned/defined anywhere. Instead, use similar wording as before (regarding default port and how to override it).